internal/pkg/rpaas/nginx/configuration_render.go

```diff
@@ -230,15 +230,26 @@ http {
     {{- end }}
     {{- end}}
 
-    {{- if $instance.Spec.Host }}
-    upstream rpaas_default_upstream {
-        server {{ $instance.Spec.Host }};
+		{{- range $index, $bind := $instance.Spec.Binds }}
 
-        {{- with $config.UpstreamKeepalive }}
-        keepalive {{ . }};
-        {{- end }}
-    }
-    {{- end }}
+			{{- if eq $index 0 }}
+				upstream rpaas_default_upstream {
+					server {{ $bind.Host }};
+
+					{{- with $config.UpstreamKeepalive }}
+					keepalive {{ . }};
+					{{- end }}
+			}
+			{{- end }}
+
+			upstream rpaas_backend_{{ $bind.Name }} {
+				server {{ $bind.Host }};
+			{{- with $config.UpstreamKeepalive }}
+			keepalive {{ . }};
+			{{- end }}
+			}
+
+		{{- end }}
```

Comment:
Please, make sure you are using regular spaces instead of tabs on this template.